misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java

```diff
@@ -312,16 +329,41 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
      * @param pkgEnvMap map of BPackageSymbol and its respective SymbolEnv
      * @param packageID Fully qualified package ID of the respective function
      * @param functionName Name of the function
+     * @param objectName Name of the object
      * @return Function type if found, null if not found
      */
-    private BType getFunctionType(Map<BPackageSymbol, SymbolEnv> pkgEnvMap, PackageID packageID, String functionName) {
+    private BType getFunctionType(Map<BPackageSymbol, SymbolEnv> pkgEnvMap, PackageID packageID, String functionName,
+                                  String objectName) {
         // Symbol resolver, Pass the acquired package Symbol from package cache
         for (Map.Entry<BPackageSymbol, SymbolEnv> entry : pkgEnvMap.entrySet()) {
             // Multiple packages may be present with same name, so all entries must be checked
             if (entry.getKey().pkgID.equals(packageID)) {
-                BSymbol symbol = symbolResolver.lookupSymbolInMainSpace(entry.getValue(), new Name(functionName));
-                if (!symbol.getType().toString().equals("other")) {
-                    return symbol.getType();
+                SymbolEnv entrySymbolEnv = entry.getValue();
+
+                // If the object name is specified then you have to exclusively check the entrySymbolEnv for the enclPkg
+                if (objectName != null) {
+                    BLangPackage enclPkg = entrySymbolEnv.enclPkg;
+                    List<BLangTypeDefinition> typeDefinitionList = enclPkg.getTypeDefinitions();
+
+                    // Look thru all the existing definitions for objectName
+                    for (BLangTypeDefinition objectBlangType : typeDefinitionList) {
+                        // If the correct type definition is found, we need to look thru its list of function
+                        if (objectBlangType.getName().toString().equals(objectName)) {
+                            List<BLangFunction> objectMethods =
+                                    ((BLangObjectTypeNode) objectBlangType.typeNode).getFunctions();
+                            // Check every method for the matching function name
+                            for (BLangFunction objectMethod : objectMethods) {
+                                if (objectMethod.getName().toString().equals(functionName)) {
```

Comment:
Can we reduce the depth? 

Also can we cache the funcNames, without iterating through everything every-time?

Reply:
Im trying to reduce the program depth by generating a map with all of the object types and associated functions. 
This way I can just pass the Object name and the function name to a function and it can return the required type. Would this implementation suffice?

Reply:
I have refactored the code to introduce the concept I have mentioned above.